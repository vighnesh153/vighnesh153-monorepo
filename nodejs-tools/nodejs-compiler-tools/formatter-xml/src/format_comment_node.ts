import { XmlCommentNode } from '@vighnesh153/parser-xml';
import { buildIndentationSpace } from './build_indentation_space.ts';

type FormatCommentNodeOptions = {
  commentNode: XmlCommentNode;
  indentationLevel: number;
  indentation: number;
};

export function formatCommentNode({ commentNode, indentationLevel, indentation }: FormatCommentNodeOptions): string {
  return (
    buildIndentationSpace({ indentationLevel, indentation }) + `<!-- ${commentNode.comment.tokenLiteral.trim()} -->`
  );
}
