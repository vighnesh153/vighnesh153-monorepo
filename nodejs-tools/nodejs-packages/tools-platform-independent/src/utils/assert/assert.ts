import { not } from '../not';

export function assert(condition: boolean, message = 'Assertion failed') {
  if (not(condition)) {
    throw new Error(message);
  }
}
