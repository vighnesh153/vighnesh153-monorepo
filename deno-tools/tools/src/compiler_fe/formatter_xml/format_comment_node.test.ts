import { assertEquals } from "@std/assert";
import { assertSnapshot } from "@std/testing/snapshot";
import type { XmlCommentNode } from "@/compiler_fe/parser_xml/mod.ts";
import { formatCommentNode } from "./format_comment_node.ts";
import { parseProgram } from "./test_utils.ts";

Deno.test("formatCommentNode should format comment node with 0 indentation level", async (t) => {
  const [parser, program] = parseProgram(
    `<!-- <uses-permission android:name="com.android.providers.tv.permission.WRITE_EPG_DATA" /> -->`,
  );

  assertEquals(parser.errors.length, 0);
  assertEquals(program.statements.length, 1);
  await assertSnapshot(
    t,
    formatCommentNode({
      commentNode: program.statements[0] as XmlCommentNode,
      indentation: 4,
      indentationLevel: 0,
    }),
  );
});

Deno.test("formatCommentNode should format comment node with 2 indentation level", async (t) => {
  const [parser, program] = parseProgram(
    `<!-- <uses-permission android:name="com.android.providers.tv.permission.WRITE_EPG_DATA" /> -->`,
  );

  assertEquals(parser.errors.length, 0);
  assertEquals(program.statements.length, 1);
  await assertSnapshot(
    t,
    formatCommentNode({
      commentNode: program.statements[0] as XmlCommentNode,
      indentation: 4,
      indentationLevel: 2,
    }),
  );
});

Deno.test("formatCommentNode should format comment node with 2 indentation level and 3 indentation", async (t) => {
  const [parser, program] = parseProgram(
    `<!-- <uses-permission android:name="com.android.providers.tv.permission.WRITE_EPG_DATA" /> -->`,
  );

  assertEquals(parser.errors.length, 0);
  assertEquals(program.statements.length, 1);
  await assertSnapshot(
    t,
    formatCommentNode({
      commentNode: program.statements[0] as XmlCommentNode,
      indentation: 3,
      indentationLevel: 2,
    }),
  );
});
