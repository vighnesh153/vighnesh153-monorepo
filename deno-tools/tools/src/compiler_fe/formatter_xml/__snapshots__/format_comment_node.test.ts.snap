export const snapshot = {};

snapshot[`formatCommentNode should format comment node with 0 indentation level 1`] = `'<!-- <uses-permission android:name="com.android.providers.tv.permission.WRITE_EPG_DATA" /> -->'`;

snapshot[`formatCommentNode should format comment node with 2 indentation level 1`] = `'        <!-- <uses-permission android:name="com.android.providers.tv.permission.WRITE_EPG_DATA" /> -->'`;

snapshot[`formatCommentNode should format comment node with 2 indentation level and 3 indentation 1`] = `'      <!-- <uses-permission android:name="com.android.providers.tv.permission.WRITE_EPG_DATA" /> -->'`;
